include/GafferScene/TweakPlug.inl

```diff
@@ -52,6 +78,102 @@ const T *TweakPlug::valuePlug() const
 	return IECore::runTimeCast<const T>( valuePlugInternal() );
 }
 
+template<class GetDataFunctor, class SetDataFunctor>
+bool TweakPlug::applyTweak(
+	GetDataFunctor &&getDataFunctor,
+	SetDataFunctor &&setDataFunctor,
+	MissingMode missingMode
+) const
+{
+	if( !enabledPlug()->getValue() )
+	{
+		return false;
+	}
+
+	const std::string name = namePlug()->getValue();
+	if( name.empty() )
+	{
+		return false;
+	}
+
+	const Mode mode = static_cast<Mode>( modePlug()->getValue() );
+
+	if( mode == GafferScene::TweakPlug::Remove )
+	{
+		return setDataFunctor( name,  nullptr );
+	}
+
+	const IECore::Data *currentValue = getDataFunctor( name );
+	IECore::DataPtr newData = Gaffer::PlugAlgo::getValueAsData( valuePlug() );
+	if( !newData )
+	{
+		throw IECore::Exception(
+			boost::str( boost::format( "Cannot apply tweak to \"%s\" : Value plug has unsupported type \"%s\"" ) % name % valuePlug()->typeName() )
+		);
+	}
+	if( currentValue && currentValue->typeId() != newData->typeId() )
+	{
+		throw IECore::Exception(
+			boost::str( boost::format( "Cannot apply tweak to \"%s\" : Value of type \"%s\" does not match parameter of type \"%s\"" ) % name % currentValue->typeName() % newData->typeName() )
+		);
+	}
+
+	if( !currentValue )
+	{
+		if( missingMode == GafferScene::TweakPlug::MissingMode::Ignore )
+		{
+			return false;
+		}
+		else if( !( mode == GafferScene::TweakPlug::Replace && missingMode == GafferScene::TweakPlug::MissingMode::IgnoreOrReplace) )
+		{
+			throw IECore::Exception( boost::str( boost::format( "Cannot apply tweak with mode %s to \"%s\" : This parameter does not exist" ) % Detail::modeToString( mode ) % name ) );
+		}
+	}
+
+	if(
+		mode == GafferScene::TweakPlug::Add ||
+		mode == GafferScene::TweakPlug::Subtract ||
+		mode == GafferScene::TweakPlug::Multiply
+	)
+	{
+		modifyData( currentValue, newData.get(), newData.get(), mode, name );
+	}
+
+	setDataFunctor( name, newData );
+
+	return true;
+}
+
+template<class GetDataFunctor, class SetDataFunctor>
+bool TweaksPlug::applyTweaks(
+	GetDataFunctor &&getDataFunctor,
+	SetDataFunctor &&setDataFunctor,
+	TweakPlug::MissingMode missingMode
+) const
+{
+	bool tweakApplied = false;
+
+	for( auto tweakPlug : TweakPlug::Range( *this ) )
```

Comment:
I think we want `auto &` here. The type we're iterating over is `TweakPlugPtr`, and we don't want to incur the extra reference counting operations that a copy would entail.

Reply:
Fixed in 9b794ef88ab1ddd6435bc923e50cc3df76017da3